### PR TITLE
feature/adding_error_code_for_proxy_connection_issues

### PR DESCRIPTION
**Pull request by cornillejr**
proxy_error.html should be deployed under /usr/share/nginx/html (at the root of the nginx web folder).

@jbutkus @ewanmttk Could you please review?

### CODE REVIEW

File: proxy_error.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?><methodResponse><params><param><value><struct><member><name>resultCode</name><value><int>998</int></value></member></struct></value></param></params></methodResponse>
```
Comment (jbutkus) [inline proxy_error.html]: Would you think a readable layout should be considered bad?

```suggestion
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<methodResponse>
    <params>
        <param>
            <value>
                <struct>
                    <member>
                        <name>resultCode</name>
                        <value>
                            <int>998</int>
                        </value>
                    </member>
                </struct>
            </value>
        </param>
    </params>
</methodResponse>
```

File: nginx_conf.txt [hunk 2]
```diff
@@ -60,6 +60,9 @@ http {
             proxy_connect_timeout   3s;
             proxy_send_timeout      2s;
             proxy_ssl_session_reuse on;
+            proxy_intercept_errors on;
```
Comment (jbutkus) [inline nginx_conf.txt]: Alignment??? 😱 
```suggestion
            proxy_intercept_errors  on;
```

File: nginx_conf.txt [hunk 3]
```diff
@@ -60,6 +60,9 @@ http {
             proxy_connect_timeout   3s;
             proxy_send_timeout      2s;
             proxy_ssl_session_reuse on;
+            proxy_intercept_errors on;
+
+            error_page 404 500 502 503 504 =200 /proxy_error.html;
```
Comment (jbutkus) [inline nginx_conf.txt]: Why do you want to mask `404`?
I think "endpoint missing" shouldn't be ignored this easily?

Comment (cornillejr) [reply nginx_conf.txt]: @jbutkus All those error codes mean the same thing as far as we are concerned, the endpoint isn't accessible. The whole point of this is to separate downstream errors from upstream ones.

**Pull request merged**